pkg/image/fetcher.go

```diff
@@ -106,6 +107,22 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 		if err == nil || !errors.Is(err, ErrNotFound) {
 			return img, err
 		}
+	case PullWithInterval, PullDaily, PullHourly, PullWeekly:
+		img, err := f.fetchDaemonImage(name)
```

Comment:
Shouldn't we check the interval first before deciding to fetch from the daemon? 

```go
if pull, _ := f.CheckImagePullInterval(name); !pull {
   img, err := f.fetchDaemonImage(name)
   if err == nil {
      // removes it from the image.json file, maybe it was deleted
   }
   return img, err
} 
```

Reply:
I think it's related to this [message](https://github.com/buildpacks/pack/issues/1368#issuecomment-1956723355)
I will update it soon.